### DISCUSSION TITLE
Album Covers Automatically Changed to 96DPI

### DISCUSSION THREAD

**Discussion by heathcliff2019** [AzuraCast/AzuraCast#7537] [General]
Hello,

I'm experiencing an issue with album covers being automatically changed from 72DPI to 96DPI. Here are the details:

Issue: 

- I replace album covers with a 72dpi, 250x250px, JPG (quality 70%), but Azura is changing it back to 96DPI.
- Impact: This change is unnecessary for online display and adds extra work for the system without any apparent benefit.

Additional Issues:

1. I noticed playback issues in the log file indicating that some tracks are not being played due to an issue with the cover.
2. There have been issues with WEBP files and covers' DPI.
3. It seems AzuraCast is automatically reverting covers back to 96DPI, which is problematic for online radio stations and web usage because higher DPI values are not required for web display and can cause unnecessary strain on system resources.

Steps to Reproduce:

1. Upload a 72dpi, 250x250px, JPG (quality 70%) album cover.
2. Save and check the album cover settings.
3. Notice that the DPI is automatically changed to 96DPI.

- Expected Behavior: The DPI should remain at 72 as originally set.
- Actual Behavior: The DPI is changed to 96.

I just want to bring awareness to this issue.

Thank you for your time reading this post.

**Comment by RM-FM** [comment]
I believe DPI doesn't matter for web use. It's irrelevant for web-based or on-screen images since digital displays measure images in pixels, not inches. The image is specified as 250x250 pixels, which defines the actual on-screen size regardless of DPI. DPI is generally only relevant for print, where it determines the printed size of the image.

The expectation that DPI should remain at 72 indicates that the you want the metadata to remain unchanged, even though this metadata typically has no practical impact on how a digital image is displayed.

**Reply by heathcliff2019** [reply]
Yes, DPI doesn’t affect web display, but that’s not the point here. The issue is that AzuraCast is overriding settings you intentionally set, without any good reason. When you upload at 72 DPI, you expect it to stay that way. Changing it to 96 DPI doesn’t improve anything and just messes with consistency. On top of that, AzuraCast is using resources making these adjustments. Every time it changes DPI, it’s using processing power that could be better spent on streaming. With multiple files, this adds up and could even slow down performance over time.

**Reply by RM-FM** [reply]
I believe the opposite is true. Azuracast doesn’t modify the DPI metadata at all, which is why the `gd` image library defaults it to `96`. If processing power is your main concern, then setting the DPI to the correct value (or hard-coding it to `72`) would actually add more work for the system - even though, it likely won’t make a noticeable difference.

**Reply by heathcliff2019** [reply]
Thank you very much for your input and thinking with me.

So ..previously, I never adjusted the DPI of my album covers. The covers were set to 96 DPI by default, and everything worked seamlessly. It was only when playback issues started to appear, and logs indicated that the cover was the cause, that I decided to test lowering the DPI to 72 as a troubleshooting step. This means the issue wasn’t triggered by my settings but rather by how AzuraCast handles cover metadata.

Your suggestion to simply stop adjusting the DPI misses the point. Adjusting to 72 DPI was a reactive measure to address issues that didn’t exist with the same files at 96 DPI before. So the problem doesn’t stem from my workflow but from AzuraCast’s handling of cover metadata. Ideally, default settings should work reliably without impacting playback.

It appears AzuraCast is performing an automatic DPI adjustment, which is unnecessary. This adjustment should simply be left out, allowing the original DPI to be retained. This way, users like myself can maintain a consistent workflow without forced adjustments or unnecessary system load.

I’d like to point out that platforms like Radio.co, MediaCP, and Centova Cast don't have this issues. This makes the issue unique to AzuraCast and causes unnecessary problems that don’t arise in other systems.

It’s uncommon to set album covers for web use specifically at 96 DPI. DPI mainly applies to print, where dots per inch determine the quality of the printed image. For web, only pixel dimensions (e.g., 250x250 px) determine the size and clarity on screen. Enforcing a specific DPI value (like 96 DPI) for online album covers that appear in small formats is totally unnecessary and inefficient.

**Comment by BusterNeece** [comment]
Looking at our code...no, nothing we're doing is setting the DPI of album art to 96 DPI. As you've mentioned, there's no reason for us to do so anyway since we're expecting that the art will be displayed for the web where DPI doesn't really matter.

It's possible that the GD library itself is trying to change the DPI, but in my own local tests, when I upload a 72 DPI image in the Rolling Release, it spits out a 72 DPI image.

If you're able to reliably reproduce the issue with a specific image (i.e. you have a 72dpi image, upload it to AzuraCast and it spits out a 96dpi one), then I'd recommend creating a new GitHub Issue for that, as that's a bug and not intended behavior.
